### PR TITLE
Remove extra HdfsConfiguration#getConfiguration method

### PR DESCRIPTION
**Pull request by arhimondr**
Create configuration in KerberosHadoopAuthentication using HdfsConfigurationUpdater

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveHdfsConfiguration.java [hunk 1]
```diff
@@ -21,23 +21,13 @@
...
-        Configuration.addDefaultResource("hdfs-default.xml");
-        Configuration.addDefaultResource("hdfs-site.xml");
-
-        // must not be transitively reloaded during the future loading of various Hadoop modules
-        // all the required default resources must be declared above
-        INITIAL_CONFIGURATION = new Configuration(false);
-        Configuration defaultConfiguration = new Configuration();
-        copy(defaultConfiguration, INITIAL_CONFIGURATION);
-    }
+    private static final Configuration INITIAL_CONFIGURATION = getInitialConfiguration();
```
Comment (kokosing) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveHdfsConfiguration.java]: inline

File: presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHadoopAuthentication.java [hunk 2]
```diff
@@ -28,20 +30,25 @@
...
 
+    private static Configuration getConfiguration(HdfsConfigurationUpdater updater)
+    {
+        Configuration configuration = new Configuration(false);
+        copy(getInitialConfiguration(), configuration);
+        updater.updateConfiguration(configuration);
+        return configuration;
+    }
+
     private static void initializeConfigs(Configuration configuration)
```
Comment (kokosing) [inline presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHadoopAuthentication.java]: shouldn't be `configuration.set("hadoop.security.authentication", "kerberos");` this moved to hdfs conf updater now?

File: presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHadoopAuthentication.java [hunk 3]
```diff
@@ -28,20 +30,25 @@
...
 
-    public static KerberosHadoopAuthentication createKerberosHadoopAuthentication(KerberosAuthentication kerberosAuthentication, HdfsConfiguration hdfsConfiguration)
+    public static KerberosHadoopAuthentication createKerberosHadoopAuthentication(KerberosAuthentication kerberosAuthentication, HdfsConfigurationUpdater updater)
     {
-        KerberosHadoopAuthentication kerberosHadoopAuthentication = new KerberosHadoopAuthentication(kerberosAuthentication);
-
-        Configuration configuration = requireNonNull(hdfsConfiguration, "hdfsConfiguration is null").getConfiguration();
-        initializeConfigs(configuration);
-        return kerberosHadoopAuthentication;
+        initializeConfigs(getConfiguration(updater));
```
Comment (kokosing) [inline presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHadoopAuthentication.java]: I would inline `initializeConfigs` (if it is used in single place) and `getConfiguration` here

File: presto-hive/src/main/java/com/facebook/presto/hive/HdfsConfiguration.java [hunk 4]
```diff
@@ -20,6 +20,5 @@
 
 public interface HdfsConfiguration
 {
-    Configuration getConfiguration();
```
Comment (findepi) [inline presto-hive/src/main/java/com/facebook/presto/hive/HdfsConfiguration.java]: 👍

File: presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHadoopAuthentication.java (hunk 2 above)
Comment (findepi) [reply presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHadoopAuthentication.java]: why?

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveHdfsConfiguration.java (hunk 1 above)
Comment (arhimondr) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveHdfsConfiguration.java]: `getInitialConfiguration()` creates a copy underneath. I decided to store it here for better performance.

File: presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHadoopAuthentication.java (hunk 2 above)
Comment (arhimondr) [reply presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosHadoopAuthentication.java]: This is a really tricky property. It is statically interpreted by the `UserGroupInformation`, what is really ugly. I would really love to keep everything related to static initialization here.

**Pull request merged**